train.py

```diff
@@ -137,7 +137,7 @@ def main(job_config: JobConfig):
     # loss function to be shared by Pipeline Parallel and SPMD training
     def loss_fn(pred, labels):
         return torch.nn.functional.cross_entropy(
-            pred.flatten(0, 1), labels.flatten(0, 1)
+            pred.flatten(0, 1).float(), labels.flatten(0, 1)
```

Comment:
cc: @Chillee `torch.compile` should respect this upcast numerically?

Reply:
yeah definitely